src/pybind/mgr/diskprediction/module.py

```diff
@@ -0,0 +1,346 @@
+"""
+A diskprediction module
+"""
+from __future__ import absolute_import
+
+from datetime import datetime
+import json
+from mgr_module import MgrModule
+from threading import Event
+
+from .common import DP_MGR_STAT_ENABLED, DP_MGR_STAT_DISABLED
+from .task import MetricsTask, PredictionTask, SmartTask
+
+
+DP_TASK = [MetricsTask, PredictionTask, SmartTask]
+
+
+class Module(MgrModule):
+
+    config = dict()
+
+    OPTIONS = [
+        {
+            'name': 'diskprediction_config_mode',
+            'default': 'cloud'
+        },
+        {
+            'name': 'diskprediction_server',
+            'default': None
+        },
+        {
+            'name': 'diskprediction_port',
+            'default': 8086
+        },
+        {
+            'name': 'diskprediction_database',
+            'default': 'telegraf'
+        },
+        {
+            'name': 'diskprediction_user',
+            'default': None
+        },
+        {
+            'name': 'diskprediction_password',
+            'default': None
+        },
+        {
+            'name': 'diskprediction_cluster_domain_id',
+            'default': None
+        },
+        {
+            'name': 'diskprediction_upload_metrics_interval',
+            'default': 600
+        },
+        {
+            'name': 'diskprediction_upload_smart_interval',
+            'default': 43200
+        },
+        {
+            'name': 'diskprediction_retrieve_prediction_interval',
+            'default': 43200
+        },
+        {
+            'name': 'diskprediction_smart_relied_device_health',
+            'default': 'true'
+        }
+    ]
+
+    COMMANDS = [
+        {
+            'cmd': 'diskprediction config-mode '
+                   'name=mode,type=CephString,req=true',
+            'desc': 'config disk prediction mode [\"cloud\"|\"onpremise\"|\"local\"]',
+            'perm': 'rw'
+        },
+        {
+            'cmd': 'diskprediction set-smart-relied-device-health '
+                   'name=relied,type=CephString,req=true',
+            'desc': 'Set device health data from device health plugin',
+            'perm': 'rw'
+        },
+        {
+            'cmd': 'diskprediction config-show',
+            'desc': 'Prints diskprediction configuration',
+            'perm': 'r'
+        },
+        {
+            'cmd': 'diskprediction config-set '
+                   'name=server,type=CephString,req=true '
+                   'name=user,type=CephString,req=true '
+                   'name=password,type=CephString,req=true '
+                   'name=port,type=CephInt,req=false ',
+            'desc': 'Configure Disk Prediction service',
+            'perm': 'rw'
+        },
+        {
+            'cmd': 'diskprediction get-predicted-status '
+                   'name=osd_id,type=CephString,req=true',
+            'desc': 'Get osd physical disk predicted result',
+            'perm': 'r'
+        },
+        {
+            'cmd': 'diskprediction run-metrics-forced',
```

Comment:
For the commands which are (I think?) primarily there for debugging, perhaps adopt a prefix pattern like "diskprediction debug run-preduction".  That way it's obvious to users (plus testers, people writing docs) which commands are the user interface, and which are just for special occasions.